config/v1/types_cluster_version.go

```diff
@@ -106,6 +111,48 @@ type ClusterVersionStatus struct {
 	AvailableUpdates []Update `json:"availableUpdates"`
 }
 
+// UpdateState is a constant representing whether an update was successfully
+// applied to the cluster or not.
+type UpdateState string
+
+const (
+	// CompletedUpdate indicates an update was successfully applied
+	// to the cluster (all resource updates were successful).
+	CompletedUpdate UpdateState = "Completed"
+	// PartialUpdate indicates an update was never completely applied
+	// or is currently being applied.
+	PartialUpdate UpdateState = "Partial"
+)
+
+// UpdateHistory is a single attempted update to the cluster.
+type UpdateHistory struct {
+	// state reflects whether the update was fully applied. The Partial state
+	// indicates the update is not fully applied, while the Completed state
+	// indicates the update was successfully rolled out at least once (all
+	// parts of the update successfully applied).
+	State UpdateState `json:"state"`
+
+	// startedTime is the time at which the update was started.
+	StartedTime metav1.Time `json:"startedTime"`
+	// completionTime, if set, is when the update was fully applied. The update
+	// that is currently being applied will have a null completion time.
+	// Completion time will always be set for entries that are not the current
+	// update (usually to the started time of the next update).
+	CompletionTime *metav1.Time `json:"completionTime"`
+
+	// version is a semantic versioning identifying the update version. When this
+	// field is part of spec, version is optional if payload is specified.
+	//
+	// +optional
+	Version string `json:"version"`
+	// payload is a container image location that contains the update. When this
+	// field is part of spec, payload is optional if version is specified and the
```

Comment:
even though it's not technically required, don't we always know this?

Reply:
It is possible for a release to not have a payload version defined (version must be a semantic version), or for us not to be able to retrieve the payload by image and thus get access to the version number (image -> version)

Reply:
Updated with the different godoc (payload is required on update history because a spec update that has version, no payload, and is missing availableUpdate would not result in an update history item).